src/session_impl.cpp

```diff
@@ -5227,7 +5217,16 @@ namespace aux {
 		INVARIANT_CHECK;
 
 		std::string const net_interfaces = m_settings.get_str(settings_pack::listen_interfaces);
-		m_listen_interfaces = parse_listen_interfaces(net_interfaces);
+		if (!net_interfaces.empty())
+		{
+			m_listen_interfaces = parse_listen_interfaces(net_interfaces);
+		}
+		else
+		{
+			// if no listen interfaces are specified, create sockets to use
+			// any interface
+			m_listen_interfaces = parse_listen_interfaces("0.0.0.0:0,[::]:0");
```

Comment:
I think the `duplex::only_outgoing` flag in the previous code is relevant.

Reply:
I seem to recall that the old behavior tried to mimic, as closely as possible, not having any listen sockets. But I think the name is a bit misleading as it is. Since we use these for outgoing connections also. And I think it makes sense to preserve this attempt to not accept incoming connections for an empty string.

But doesn't this still raise the question of what happens when a torrent attempts to announce when there are literally no sockets to use?